tests/test_queuing.py

```diff
@@ -406,22 +344,119 @@ def __init__(self, *args):
                 self.queue()
 
             def queue(self):
-                qml.QueuingContext.append(self, owns=tuple(self.obs))
+                QueuingContext.append(self, owns=tuple(self.obs))
 
                 for o in self.obs:
                     try:
-                        qml.QueuingContext.update_info(o, owner=self)
+                        QueuingContext.update_info(o, owner=self)
                     except AttributeError:
                         pass
 
                 return self
 
-        with qml._queuing.AnnotatedQueue() as q:
+        with AnnotatedQueue() as q:
             A = qml.PauliZ(0)
             B = qml.PauliY(1)
             tensor_op = AnnotatingTensor(A, B)
 
         assert q.queue == [A, B, tensor_op]
-        assert q.get_info(A) == {"owner": tensor_op}
-        assert q.get_info(B) == {"owner": tensor_op}
-        assert q.get_info(tensor_op) == {"owns": (A, B)}
+        assert q._get_info(A) == {"owner": tensor_op}
+        assert q._get_info(B) == {"owner": tensor_op}
+        assert q._get_info(tensor_op) == {"owns": (A, B)}
+
+
+class TestOperationRecorder:
+    """Test the OperationRecorder class."""
+
+    def test_circuit_integration(self):
+        """Tests that the OperationRecorder integrates well with the
+        core behaviour of PennyLane."""
+        expected_output = (
+            "Operations\n"
+            + "==========\n"
+            + "PauliY(wires=[0])\n"
+            + "PauliY(wires=[1])\n"
+            + "RZ(tensor(0.4, requires_grad=True), wires=[0])\n"
+            + "RZ(tensor(0.4, requires_grad=True), wires=[1])\n"
+            + "CNOT(wires=[0, 1])\n"
+            + "\n"
+            + "Observables\n"
+            + "===========\n"
+        )
+
+        dev = qml.device("default.qubit", wires=2)
+
+        @qml.qnode(dev)
+        def circuit(a, b, c):
+            qml.RX(a, wires=0)
+            qml.RY(b, wires=1)
+
+            with qml.tape.OperationRecorder() as recorder:
+                ops = [
+                    qml.PauliY(0),
+                    qml.PauliY(1),
+                    qml.RZ(c, wires=0),
+                    qml.RZ(c, wires=1),
+                    qml.CNOT(wires=[0, 1]),
+                ]
+
+            assert str(recorder) == expected_output
+            assert recorder.queue == ops
+
+            return qml.expval(qml.PauliZ(0)), qml.expval(qml.PauliZ(1))
+
+        circuit(0.1, 0.2, 0.4)
+
+    def test_template_integration(self):
+        """Tests that the OperationRecorder integrates well with the
+        core behaviour of PennyLane."""
+        expected_output = (
+            "Operations\n"
+            + "==========\n"
+            + "RZ(0, wires=[0])\n"
+            + "RZ(3, wires=[0])\n"
+            + "RZ(6, wires=[0])\n"
+            + "RZ(9, wires=[0])\n"
+            + "RZ(12, wires=[0])\n"
+            + "\n"
+            + "Observables\n"
+            + "===========\n"
+        )
+
+        def template(x):
```

Comment:
Should we use the `@template` decorator here? That seems to be the way for defining templates, or are we considering its removal?

Reply:
A good question, however I am hesitant to change these tests in this PR as it is out of scope - better to address these in a new, specific PR.